### PR TITLE
Fix benchmarks and various other lints

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

in `dev/links` we didn't require a passing CI. This makes sure, that all jobs are green again

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1202805690238892/1203348428860466/f) _(internal)_

## 🔍 What does this change?

- Fix clippy lints
    - Implement Default for Edges: 2c4c8d0
    - Remove unnecessary function calls: 95c5528
    - Ignore option_if_let_else lint: 9080e7e
    - Box vertices: f9cd57a
    - Satisfy Send bound: 48eadbb
    - Remove unfulfilled lint attribute: d40f942
    - Ignore too_many_lines lint: 8ac9462
- Fix benchmark code: a16d234
- Fix doc links: ff4a82c

## 🛡 What tests cover this?

The CI will pick all of this up

## 🎥  Demo

image

### CODE REVIEW

File: packages/graph/hash_graph/bench/benches/util.rs [hunk 1]
```diff
@@ -229,6 +228,7 @@ pub async fn seed<D, P, L, E, C>(
     }
 
     for entity_type_str in entity_types {
+        println!("creating entity type {}", entity_type_str);
```
Comment (thehabbos007) [inline packages/graph/hash_graph/bench/benches/util.rs]: I suppose this should not be here?

File: packages/graph/hash_graph/lib/graph/src/store/query/filter.rs [hunk 2]
```diff
@@ -128,6 +128,21 @@ impl<'q> Filter<'q, EntityProperties> {
         ])
     }
 
+    /// Creates a `Filter` to search for a specific entities at their latest version, identified by
+    /// its [`EntityUuid`].
+    #[must_use]
+    pub fn for_latest_entity_by_entity_uuid(entity_uuid: EntityUuid) -> Self {
```
Comment (thehabbos007) [inline packages/graph/hash_graph/lib/graph/src/store/query/filter.rs:134]: Is this being used anywhere? (i could find it used in `representative_read`, should we put it behind cfg?)

File: packages/graph/hash_graph/bench/benches/util.rs (hunk 1 above)
Comment (TimDiekmann) [inline packages/graph/hash_graph/bench/benches/util.rs:231]: I used this for debugging

```suggestion
```

**Review by thehabbos007 — COMMENTED** [review]
Couple of comments :)

File: packages/graph/hash_graph/lib/graph/src/store/query/filter.rs (hunk 2 above)
Comment (thehabbos007) [reply packages/graph/hash_graph/lib/graph/src/store/query/filter.rs:134]: (Discussed offline, this is not using ownedbyid)

**Review by thehabbos007 — APPROVED** [review]

**Pull request merged**